### ISSUE TITLE
Decide how to support `cxxLanguageStandard` in a Swift manifest.

### ISSUE THREAD

**Issue by cgrindel** [cgrindel/rules_swift_package_manager#166]
The problem is that we do not want to add it to the `cxxopts` for each target. It would override the values in the .bazelrc.  Also, I think that you do not want modules linked with different standards.

- Should we output `cxxLanguageStandard` values as we come across them?
- Should we scan all of the manifests and pick the highest one? 
- How should we tell the client? 
  - We can print the value(s) from Starlark, but they could be easily missed in the output.
  - We can analyze the manifests during `update-repos` and print a suggested `.bazelrc` entry.

Example:
- [firebase/abseil-cpp-SwiftPM](https://github.com/firebase/abseil-cpp-SwiftPM/blob/main/Package.swift#L47)

**Comment by keith** [comment]
I think you probably do want to override the global ones for each target as this could be saying "it's only compatible with this version". At the very least that's likely the only version it's tested with.

If that became a big issue cargo raze has a clever system for overriding things like this

**Comment by cgrindel** [comment]
Interesting. I thought it best practice to specify `-std=XXX` once for the whole project. It definitely would not be difficult to specify it at the target level.

**Comment by keith** [comment]
I agree with that for all your own code but yea I think for third parties like this it's more complicated. I think many bazel libraries just throw this in their Copts as well for the same reason but I'd have to double check in envoy how many do

**Comment by cgrindel** [comment]
Closing as completed.

The "decide how to support" question was effectively answered by the resolution of the user-facing follow-up #510. The working approach is:

- Users add the needed C++ flags to their own `.bazelrc` (e.g. `build --cxxopt='-std=c++14'`, plus `build --per_file_copt=.*\.mm\$@-std=c++14` for Objective-C++ files).
- `examples/vapor_example/.bazelrc` already demonstrates the pattern for projects that need it.
- Automating `.bazelrc` rewrites was considered and declined — reliably reading / updating a `.bazelrc` (with all of its `import` directives) is too brittle for the ruleset to own.

#510 was closed as completed on 2023-10-04 ("the workaround of setting `--per_file_copt` and other fixes that came out of this are good enough for me") — which settles this one too.

If the "print suggested flags at the end of the package-tool run" refinement ends up wanted, it'd be cleaner as a fresh, narrowly-scoped issue than resurrecting this one.
